### PR TITLE
Kusto PowerShell Client

### PR DESCRIPTION
**Pull request by Aviv-Yaniv**
## Checklist

- [ ] I have read the [_Submitting Changes_](../blob/master/CONTRIBUTING.md#submitting-changes) section of [`CONTRIBUTING.md`](../blob/master/CONTRIBUTING.md)
- [x] The title of the PR is clear and informative
- [ ] The appropriate `ChangeLog.md` file(s) has been updated:
    - For any service, the `ChangeLog.md` file can be found at `src/{{SERVICE}}/{{SERVICE}}/ChangeLog.md`
    - A snippet outlining the change(s) made in the PR should be written under the `## Upcoming Release` header -- no new version header should be added
- [ ] The PR does not introduce [breaking changes](../blob/master/documentation/breaking-changes/breaking-changes-definition.md)
- [ ] If applicable, the changes made in the PR have proper test coverage
- [ ] For public API changes to cmdlets:
    - [ ] a cmdlet design review was approved for the changes in [this repository](https://github.com/Azure/azure-powershell-cmdlet-review-pr) (_Microsoft internal only_)
        - {Please put the link here}
    - [ ] the markdown help files have been regenerated using the commands listed [here](../blob/master/documentation/development-docs/help-generation.md#updating-all-markdown-files-in-a-module)

### CODE REVIEW

**Comment by Aviv-Yaniv** [comment]
@wyunchi-ms Gentle ping

File: src/Kusto/README.md [hunk 1]
```diff
@@ -53,13 +53,13 @@ require:
   - $(this-folder)/../readme.azure.noprofile.md
 # lock the commit
 input-file:
-  - $(repo)/specification/azure-kusto/resource-manager/Microsoft.Kusto/stable/2021-08-27/kusto.json
-branch: 2995d5c7da8f7626449bff01d8a7d306df1e6e6d
+  - C:\GitHub\azure-rest-api-specs\specification\azure-kusto\resource-manager\Microsoft.Kusto\stable\2022-02-01\kusto.json
```
Comment (isra-fel) [inline src/Kusto/README.md:56]: Please follow the original way to specify the swagger file and its commit ID - we do not allow generating basing on a local file

File: src/Kusto/examples/Get-AzKustoOperationsResultsLocation.md [hunk 2]
```diff
@@ -0,0 +1,22 @@
+### Example 1: {{ Add title here }}
```
Comment (isra-fel) [inline src/Kusto/examples/Get-AzKustoOperationsResultsLocation.md:1]: Please fill in the examples and make sure to run `build-module.ps1` to regenerate help files

**Review by isra-fel — CHANGES_REQUESTED** [review]

File: src/Kusto/docs/Get-AzKustoOperationsResultsLocation.md [hunk 3]
```diff
@@ -0,0 +1,185 @@
...
+
+## SYNOPSIS
+Returns operation results.
+
+## SYNTAX
+
+### Get (Default)
+```
+Get-AzKustoOperationsResultsLocation -Location <String> -OperationId <String> [-SubscriptionId <String[]>]
+ [-DefaultProfile <PSObject>] [-PassThru] [<CommonParameters>]
```
Comment (isra-fel) [inline src/Kusto/docs/Get-AzKustoOperationsResultsLocation.md:18]: There seems no particular need for `-PassThru` parameter. Use [hide directive] to hide it.
Also [rename the cmdlet](https://github.com/Azure/autorest.powershell/blob/main/docs/directives.md#cmdlet-rename) from `Get-AzKustoOperationsResultsLocation` to `Get-AzKustoOperationsResultLocation` so it's consistent with `Get-AzKustoOperationsResult`.

**Review by isra-fel — CHANGES_REQUESTED** [review]
I found `Get-AzKustoOperationsResult` is also missing examples.

**Comment by Aviv-Yaniv** [comment]
Thank you @isra-fel!

Changelog:
        * Support inline script resource (creation of script with content instead of sas token)
        * Add managed identity support to EventGrid
        * Add `databaseRouting` (Single/Multi) to all data connections]
        * Add `PublicIPType` to cluster

**Comment by Aviv-Yaniv** [comment]
@wyunchi-ms Please review :)

File: src/Kusto/examples/New-AzKustoDataConnection.md [hunk 4]
```diff
@@ -1,9 +1,7 @@
 ### Example 1: Create a new EventHub data connection
 ```powershell
-New-AzKustoDataConnection -ResourceGroupName "testrg" -ClusterName "testnewkustocluster" -DatabaseName "mykustodatabase" -DataConnectionName "myeventhubdc" -Location "East US" -Kind "EventHub" -EventHubResourceId "/subscriptions/$subscriptionId/resourcegroups/testrg/providers/Microsoft.EventHub/namespaces/myeventhubns/eventhubs/myeventhub" -DataFormat "JSON" -ConsumerGroup '$Default' -Compression "None" -TableName "Events" -MappingRuleName "EventsMapping"
-```
+PS C:\> New-AzKustoDataConnection -ResourceGroupName "testrg" -ClusterName "testnewkustocluster" -DatabaseName "mykustodatabase" -DataConnectionName "myeventhubdc" -Location="East US" -Kind "EventHub" -EventHubResourceId "/subscriptions/$subscriptionId/resourcegroups/testrg/providers/Microsoft.EventHub/namespaces/myeventhubns/eventhubs/myeventhub" -DataFormat "JSON" -ConsumerGroup 'Default' -Compression "None" -TableName "Events" -MappingRuleName "EventsMapping"
```
Comment (isra-fel) [inline src/Kusto/examples/New-AzKustoDataConnection.md:3]: We recently fixed some issues in help documents, see https://github.com/Azure/azure-powershell/pull/17653 -- that's what caused the conflict. However it seems by resolving the conflicts the issues are back. For example `-Location="East US"` where `=` is invalid in PowerShell.
Also we intentionally split the input and output of each example in two code blocks. And we removed the `PS C:\>` prompt for it to be on-click runnable on docs.microsoft.com .

So please check those files with conflict again. I'm almost certain you can simply take the version on the `generation ` branch.

File: src/Kusto/examples/New-AzKustoDatabase.md [hunk 5]
```diff
@@ -1,12 +1,18 @@
-### Example 1: Create a new database
+### Example 1: {{ Add title here }}
```
Comment (isra-fel) [inline src/Kusto/examples/New-AzKustoDatabase.md:1]: This file looks like a bad merge. Content was replaced by placeholder.

**Review by isra-fel — DISMISSED** [review]

File: src/Kusto/docs/Get-AzKustoScript.md [hunk 6]
```diff
@@ -34,32 +34,23 @@ Gets a Kusto cluster database script.
 
 ## EXAMPLES
 
-### Example 1: List all Kusto cluster database scripts
-
+### Example 1: {{ Add title here }}
```
Comment (wyunchi-ms) [inline src/Kusto/docs/Get-AzKustoScript.md:37]: Please take the previous change.

File: src/Kusto/docs/New-AzKustoDatabase.md [hunk 7]
```diff
@@ -23,18 +23,23 @@ Creates or updates a database.
 
 ## EXAMPLES
 
-### Example 1: Create a new database
+### Example 1: {{ Add title here }}
```
Comment (wyunchi-ms) [inline src/Kusto/docs/New-AzKustoDatabase.md:26]: Please take the previous change.

File: src/Kusto/docs/New-AzKustoScript.md [hunk 8]
```diff
@@ -32,19 +32,23 @@ Creates a Kusto database script.
 
 ## EXAMPLES
 
-### Example 1: Create a Kusto database script
+### Example 1: {{ Add title here }}
```
Comment (wyunchi-ms) [inline src/Kusto/docs/New-AzKustoScript.md:35]: Please take the previous change.

File: src/Kusto/docs/Remove-AzKustoScript.md [hunk 9]
```diff
@@ -30,12 +30,23 @@ Deletes a Kusto principalAssignment.
 
 ## EXAMPLES
 
-### Example 1: Delete a Kusto script
+### Example 1: {{ Add title here }}
 ```powershell
-Remove-AzKustoScript -ClusterName testnewkustocluster -ResourceGroupName testrg -DatabaseName mykustodatabase -Name newkustoscript
+PS C:\> {{ Add code here }}
```
Comment (wyunchi-ms) [inline src/Kusto/docs/Remove-AzKustoScript.md:35]: Please fix the example and regenerate the doc

**Comment by Aviv-Yaniv** [comment]
@isra-fel @wyunchi-ms Aligned with comments & all checks passed

**Review by wyunchi-ms — DISMISSED** [review]

File: src/Kusto/examples/Add-AzKustoDatabasePrincipal.md [hunk 10]
```diff
@@ -1,9 +1,7 @@
 ### Example 1: Add Database principals permissions
 ```powershell
-Add-AzKustoDatabasePrincipal -ResourceGroupName testrg -ClusterName testnewkustocluster -DatabaseName mykustodatabase -Value (@{Name="Some User"; Role="Admin"; Type="User"; Email="someuser@microsoft.com"})
-```
+PS C:\> Add-AzKustoDatabasePrincipal -ResourceGroupName testrg -ClusterName testnewkustocluster -DatabaseName mykustodatabase -Value (@{Name="Some User"; Role="Admin"; Type="User"; Email="someuser@microsoft.com"})
 
-```output
```
Comment (wyunchi-ms) [inline src/Kusto/examples/Add-AzKustoDatabasePrincipal.md:6]: Please keep the `output` block and remove the `PS C:\> ` in all `powershell` block. This is our new format.

**Comment by Aviv-Yaniv** [comment]
@wyunchi-ms Aligned all as per the comment

**Review by wyunchi-ms — APPROVED** [review]

**Pull request merged**